extensions/pkg/webhook/registration.go

```diff
@@ -137,6 +146,11 @@ func buildRule(mgr manager.Manager, t runtime.Object) (*admissionregistrationv1.
 		return nil, fmt.Errorf("could not get REST mapping from GroupVersionKind '%s': %w", gvk.String(), err)
 	}
 
+	resources := []string{mapping.Resource.Resource}
```

Comment:
I think this design is a bit quirky:
Specifying 
```go
[]Type{{Obj: &corev1.Node{}, Subresources: []string{"status"}}`
```
 leads to 
```yaml
    resources:
    - nodes
    - nodes/status
```

So, you are not able to specify only `nodes/status` via this.

I propose to change the `Type` struct to 
```go
type Type struct {
	Obj         client.Object
	Subresource string
}
```

and generate exactly one `resources` entry out of each `Type`.
If one wants to specify both `nodes` and `nodes/status`, one would pass 
```go
[]Type{
  {Obj: &corev1.Node{}},
  {Obj: &corev1.Node{}, Subresource: "status"}
}
```
which is more explicit and intuitive.